src/components/ListItem.js

```diff
@@ -0,0 +1,32 @@
+import React, { useState, useEffect } from 'react';
+import { db } from '../lib/firebase.js';
+import { collection, getDocs } from 'firebase/firestore';
+
+const ListItem = () => {
+  const [items, setItems] = useState([]);
+  const itemsCollectionRef = collection(db, 'shopping-list');
+
+  // get items
+  useEffect(() => {
+    const getItems = async () => {
+      const data = await getDocs(itemsCollectionRef);
+      setItems(data.docs.map((doc) => ({ ...doc.data(), id: doc.id })));
+    };
+    getItems();
+  }, []);
```

Comment:
This really works. The data is fetched but just once. When the user adds other items to the shopping list, they don't show up immediately except if the user refreshes the browser to get the newly added items. The firebase doc suggests that we add a listener to listen for changes in the firestore which will initiate another get request to fetch the newly added items. 

This is how I implemented it locally and it works really well. You don't need to refresh the browser to see new items added.

```
const ListItem = () => {
  const [items, setItems] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);


  useEffect(() => {
    setLoading(true);
    const getItems = () => {
      try {
        const itemsCollectionRef = collection(db, 'shopping-list');
        const queryShoppingList = query(itemsCollectionRef);

        onSnapshot(queryShoppingList, (querySnapshot) => {
          const items = querySnapshot.docs.reduce((acc, doc) => {
            const { itemName, buyingTime } = doc.data();
            const id = doc.id;
            return [...acc, { id, itemName, buyingTime }];
          }, []);

          setItems(items);
          setLoading(false);
        });
      } catch (e) {
        setError(true);
      }
    };

    getItems();

  }, []);

  return (
    <div>
      {error && <p>An error occured while getting your items</p>}
      {loading && <p>Loading...</p>}
      {items.map((item) => {
        return (
          <div>
            <p>Name of Item: {item.itemName}</p>
            <p>Buying Time: {item.buyingTime}</p>
          </div>
        );
      })}
    </div>
  );
};

export default ListItem;
```

Hope this is helpful.👍👍

Reply:
Hi @Amaka202,
Thank you for the feedback! 😀

I and @fatima-ola will take look and work on this, and testing it out too 😊